test/FastTests/Client/QueriesWithCustomFunctions.cs

```diff
@@ -1667,7 +1668,230 @@ select new
+                                    {
+                                        TargetId = x.TargetId,
+                                        TargetValue = x.TargetValue
+                                    })
+                            };
+
+                        Assert.Equal("FROM Documents as d WHERE (id() IN ($p0)) AND (Deleted = $p1) " +
+                                     "SELECT { Id : d.Id, Deleted : d.Deleted, " +
+                                     "Values : d.SubDocuments.filter(function(x){return [\"id2\"].length===0||[\"id2\"].indexOf(x.TargetId)>=0;}).map(function(x){return {TargetId:x.TargetId,TargetValue:x.TargetValue};}) }"
+                            , projection.ToString());
+
+                        var result = projection.ToList();
+                        Assert.Equal(2, result.Count);
+
+                        Assert.Equal(doc1.Id, result[0].Id);
+
+                        var values = result[0].Values.ToList();
+                        Assert.Equal(1, values.Count);
+                        Assert.Equal("id2", values[0].TargetId);
+
+                        Assert.Equal(doc2.Id, result[1].Id);
+
+                        values = result[1].Values.ToList();
+                        Assert.Equal(0, values.Count);
+                    }
+                }
+            }
+
+            [Fact]
+            public void CanProjectWithStringParameter()
+            {
+                using (var store = GetDocumentStore())
+                {
+                    SetUp(store);
+
+                    using (var session = store.OpenSession())
+                    {
+                        var ids = new[] { doc1.Id, doc2.Id, doc3.Id };
+                        var targetId = "id2";
+
+                        var projection =
+                            from d in session.Query<Document>().Where(x => x.Id.In(ids))
+                            where d.Deleted == false
+                            select new
+                            {
+                                Id = d.Id,
+                                Deleted = d.Deleted,
+                                Values = d.SubDocuments
+                                    .Where(x => targetId == null || x.TargetId == targetId)
+                                    .Select(x => new Result
+                                    {
+                                        TargetId = x.TargetId,
+                                        TargetValue = x.TargetValue
+                                    })
+                            };
+
+                        Assert.Equal("FROM Documents as d WHERE (id() IN ($p0)) AND (Deleted = $p1) " +
+                                     "SELECT { Id : d.Id, Deleted : d.Deleted, " +
+                                     "Values : d.SubDocuments.filter(function(x){return \"id2\"===null||x.TargetId===\"id2\";}).map(function(x){return {TargetId:x.TargetId,TargetValue:x.TargetValue};}) }"
```

Comment:
undefined !=== null, but I guess it doesn't affect here when parameters are passed via LINQ context